src/main/java/net/fabricmc/lorenztiny/TinyMappingFormat.java

```diff
@@ -0,0 +1,155 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package net.fabricmc.lorenztiny;
+
+import net.fabricmc.mapping.reader.v2.MappingParseException;
+import net.fabricmc.mapping.tree.TinyMappingFactory;
+import net.fabricmc.mapping.tree.TinyTree;
+import org.cadixdev.lorenz.MappingSet;
+import org.cadixdev.lorenz.io.MappingFormat;
+import org.cadixdev.lorenz.io.MappingsReader;
+import org.cadixdev.lorenz.model.Mapping;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+/**
+ * A helper class for working with Tiny mappings with Lorenz,
+ * in lieu of having a {@link MappingFormat mapping format}.
+ *
+ * @author Jamie Mansfield
+ * @since 2.0.0
+ */
+public enum TinyMappingFormat {
+
+	/**
+	 * Reads tiny mappings with new standard ({@code "tiny"} header).
+	 *
+	 * @see TinyMappingFactory#load(BufferedReader)
+	 */
+	STANDARD {
+		@Override
+		protected TinyTree load(final BufferedReader reader) throws IOException, MappingParseException {
+			return TinyMappingFactory.load(reader);
+		}
+	},
+
+	/**
+	 * Reads tiny mappings with legacy format ({@code "v1"} header).
+	 *
+	 * @see TinyMappingFactory#loadLegacy(BufferedReader)
+	 */
+	LEGACY {
+		@Override
+		protected TinyTree load(final BufferedReader reader) throws IOException, MappingParseException {
+			return TinyMappingFactory.loadLegacy(reader);
+		}
+	},
+
+	/**
+	 * Reads tiny mappings of either the standard (new) or legacy format.
+	 *
+	 * @see TinyMappingFactory#loadWithDetection(BufferedReader)
+	 */
+	DETECT {
+		@Override
+		protected TinyTree load(final BufferedReader reader) throws IOException, MappingParseException {
+			return TinyMappingFactory.loadWithDetection(reader);
+		}
+	},
+	;
+
+	protected abstract TinyTree load(final BufferedReader reader) throws IOException, MappingParseException;
+
+	/**
+	 * Creates a new {@link MappingsReader mappings reader} for the
+	 * mappings contained by the path supplied.
+	 * <p>
+	 * The resulting {@link Mapping mappings} will use the from namespace
+	 * (as the obfuscated names in Lorenz) to the to namespace (as the
+	 * de-obfuscated names in Lorenz).
+	 *
+	 * @param path The path to the Tiny mappings
+	 * @param from The namespace to use in the tiny file, as Lorenz's
+	 * obfuscated names
+	 * @param to The namespace to use in the tiny file, as Lorenz's
+	 * de-obfuscated names
+	 * @return The mappings reader
+	 * @throws IOException if an I/O error occurs opening the file
+	 */
+	public MappingsReader createReader(final Path path,
+									   final String from, final String to) throws IOException, MappingParseException {
```

Comment:
What's with the indentation here?

Reply:
That's odd - must've been IntelliJ when I formatted my code (originally written in a none-tabs environment).

Reply:
So this wasn't actually an issue in my IDE - but is absolutely an issue with my IDE.

Basically it replaced the spaces (where they are used to match text) with tabs, causing it to look off (since GitHub has spaces shorter than 4 spaces) - I have replaced such tabs with spaces.